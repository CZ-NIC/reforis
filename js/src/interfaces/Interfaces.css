/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

.network .scrollable {
    display: flex;
    overflow-x: auto;
}

.network ::-webkit-scrollbar {
    height: 5px;
}

.network ::-webkit-scrollbar-thumb {
    background: #95a5a6;
    border-radius: 0.2em;
}

.network ::-webkit-scrollbar-thumb:hover {
    background: #4e5a5b;
}

.network .interface {
    display: flex;
    flex-direction: column;
    margin: 0.5rem;
    align-items: center;
    cursor: pointer;
    background: transparent;
    border: none;
    padding: 0;
}

@keyframes interfaceSelectedFade {
    from {
        -webkit-filter: none;
        filter: none;
    }
    to {
        -webkit-filter: drop-shadow(0 0 0.25rem #00a2e2);
        filter: drop-shadow(0 0 0.25rem #00a2e2);
    }
}

.network .interface-selected svg {
    animation-name: interfaceSelectedFade;
    animation-duration: 0.3s;
    -webkit-filter: drop-shadow(0 0 0.5rem #00a2e2);
    filter: drop-shadow(0 0 0.25rem #00a2e2);
}

.network .interface-selected .fa-inverse {
    display: none;
}

.interface:focus {
    outline: none;
}

.interface:active {
    color: inherit;
}
