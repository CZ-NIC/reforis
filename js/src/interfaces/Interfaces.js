/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import { ForisForm } from "foris";
import API_URLs from "common/API";

import "./Interfaces.css";
import InterfacesForm from "./InterfacesForm";
import { NETWORKS_TYPES } from "./constants";

Interfaces.propTypes = {
    ws: PropTypes.object.isRequired,
};

export default function Interfaces({ ws }) {
    return (
        <>
            <h1>{_("Network Interfaces")}</h1>
            <p>
                {_(`
Here you can configure the settings of the network interfaces on your device. You can switch the physical
network interfaces among networks. If you are unsure what to set here use the default settings.
        `)}
            </p>
            <ForisForm
                ws={ws}
                forisConfig={{
                    endpoint: API_URLs.interfaces,
                    wsModule: "networks",
                }}
                prepDataToSubmit={prepDataToSubmit}
                validator={validateInterfaces}
            >
                <InterfacesForm />
            </ForisForm>
        </>
    );
}

function validateInterfaces(formData) {
    if (formData.networks.lan.length < 1) {
        return {
            networks: {
                lan: _(
                    "You have to assign at least one interface to this group."
                ),
            },
        };
    }
    return undefined;
}

function prepDataToSubmit(formData) {
    const data = { networks: {}, firewall: formData.firewall };
    NETWORKS_TYPES.forEach((network) => {
        data.networks[network] = [];
    });
    NETWORKS_TYPES.forEach((network) => {
        formData.networks[network]
            .filter((i) => i.configurable)
            .forEach((i) => data.networks[network].push(i.id));
    });
    return data;
}
