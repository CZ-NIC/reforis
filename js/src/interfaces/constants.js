/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

export const NETWORKS_CHOICES = {
    wan: _("WAN"),
    lan: _("LAN"),
    guest: _("Guest Network"),
    none: _("Unassigned"),
};
export const NETWORKS_TYPES = ["wan", "lan", "guest", "none"];

export const BUSES = ["eth", "pci", "usb", "sdio", "sfp"];
export const INTERFACE_TYPES = {
    eth: "eth",
    wifi: "wifi",
    wwan: "wwan",
};
export const INTERFACE_STATES = {
    up: "up",
    down: "down",
};
