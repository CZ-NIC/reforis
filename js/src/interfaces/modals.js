/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';

import {Modal, ModalBody, ModalFooter, ModalHeader} from '../common/bootstrap/Modal';
import Button from '../common/bootstrap/Button';


export function OpenPortsModals({shown, setShown, onKeepPortsClosed}) {
    const message = _(`
You don't have any interface assigned to the LAN network.<br/>
Do you want to <strong>open ports 22, 80 and 443 on WAN network</strong> in order to be able to access the configuration
interface of our device?                
    `);

    return <Modal shown={shown}>
        <ModalHeader setShown={setShown} title={_('Warning!')}/>
        <ModalBody>
            <p dangerouslySetInnerHTML={{__html: message}}/>
        </ModalBody>
        <ModalFooter>
            <Button className='btn-danger' onClick={onKeepPortsClosed}>{_('Keep closed')}</Button>
            <Button>{_('Open ports')}</Button>
        </ModalFooter>
    </Modal>
}


export function ConfirmModal({shown, setShown}) {
    const message = _(`
In the setup you provided it is <strong>not possible to access the administration interface</strong> of your device.
This means that the only way to configure your device will be via serial cabel or you need to a perform factory reset.
<br/><br/> 
<p><strong>Is this really what you want?</strong></p>           
`);

    return <Modal shown={shown}>
        <ModalHeader setShown={setShown} title={_('Warning!')}/>
        <ModalBody>
            <p dangerouslySetInnerHTML={{__html: message}}/>
        </ModalBody>
        <ModalFooter>
            <Button onClick={e => {
                e.preventDefault();
                setShown(false)
            }}>
                {_('Reconsider')}
            </Button>
            <Button className='btn-danger'>{_('I\'m an expert')}</Button>
        </ModalFooter>
    </Modal>
}
