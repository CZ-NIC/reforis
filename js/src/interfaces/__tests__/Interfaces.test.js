/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import {render, act, wait, getByText, fireEvent, getByLabelText} from 'react-testing-library';

import {mockedWS} from '../../testUtils/mockWS';
import mockAxios from 'jest-mock-axios';
import Interfaces from '../Interfaces';
import {interfacesFixture} from './__fixtures__/interfaces';


describe('<Interfaces/>', () => {
    let interfacesContainer;

    beforeEach(async () => {
        const mockWebSockets = new mockedWS();
        const {container} = render(<Interfaces ws={mockWebSockets}/>);
        mockAxios.mockResponse({data: interfacesFixture()});
        await wait(() => getByText(container, 'LAN1'));
        interfacesContainer = container
    });

    it('Snapshot.', () => {
        expect(interfacesContainer).toMatchSnapshot();
    });

    it('Snapshot select interface.', () => {
        act(() => {
            fireEvent.click(getByText(interfacesContainer, 'LAN1'));
        });
        expect(interfacesContainer).toMatchSnapshot();
    });

    it('Snapshot after interface moving.', async () => {
        await act(async () => {
            await fireEvent.click(getByText(interfacesContainer, 'LAN1'));
            fireEvent.change(getByLabelText(interfacesContainer, 'Network'),{target: {value: 'lan'}});
        });
        expect(interfacesContainer).toMatchSnapshot();
    });
});
