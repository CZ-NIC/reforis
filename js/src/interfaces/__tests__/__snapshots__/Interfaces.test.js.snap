// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Interfaces/> Snapshot after interface moving. 1`] = `
<div>
  <h1>
    Network Interfaces
  </h1>
  <p>
    
Here you can configure the settings of the network interfaces on your device. You can switch the physical
network interfaces among networks. If you are unsure what to set here use the default settings.
        
  </p>
  <div
    class="col-sm-12 offset-lg-1 col-lg-10 p-0 mb-3"
  >
    <form>
      <h2>
        WAN
      </h2>
      <p>
        
It acts as an external network connection. Firewall rules should be applied here. It can only contain a
single interface.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                />
              </span>
            </div>
            <h5>
              WAN
            </h5>
          </button>
        </div>
      </div>
      <h2>
        LAN
      </h2>
      <p>
        
It acts as a local network connection. LAN should contain devices which are under your control and you
trust them. These devices can see each other and can access this web interface. It is recommended that the
LAN should contain at least one interface otherwise you might not be able to configure this device in an
easy way.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN3
            </h5>
          </button>
          <button
            class="interface interface-selected"
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN1
            </h5>
          </button>
        </div>
      </div>
      <h2>
        Guest Network
      </h2>
      <p>
        
It acts as a local network connection. Unlike LAN the devices in the guest network can't access
the configuration interface of this device and are only able to access WAN (internet). This network should
be used for devices which you don't fully trust. Note that you can also limit download/upload speed of the
devices connected to the guest network.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN4
            </h5>
          </button>
        </div>
      </div>
      <h2>
        Unassigned
      </h2>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN0
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN2
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div
              style="color: gray;"
            >
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="fas fa-wifi fa-stack-1x"
                />
                <i
                  class="fas fa-slash fa-stack-1x fa-inverse"
                />
                <i
                  class="fas fa-slash fa-stack-1x"
                  style="bottom: 0.3rem;"
                />
              </span>
            </div>
            <h5>
              1
            </h5>
          </button>
        </div>
      </div>
      <h3>
        Interface %s %s
      </h3>
      <div
        class="form-group"
      >
        <label
          for="1"
        >
          Network
        </label>
        <select
          class="custom-select"
          id="1"
        >
          <option
            value="lan"
          >
            LAN
          </option>
          <option
            value="guest"
          >
            Guest Network
          </option>
          <option
            value="none"
          >
            Unassigned
          </option>
        </select>
      </div>
      <table
        class="table table-hover"
      >
        <tbody>
          <tr>
            <th>
              State
            </th>
            <td>
              <span>
                <i
                  class="fa fa-lg fa-times-circle text-danger"
                  title="down"
                />
              </span>
            </td>
          </tr>
          <tr>
            <th>
              Module ID
            </th>
            <td>
              0
            </td>
          </tr>
          <tr>
            <th>
              Slot
            </th>
            <td>
              LAN1
            </td>
          </tr>
          <tr>
            <th>
              Interface ID
            </th>
            <td>
              lan1
            </td>
          </tr>
          <tr>
            <th>
              Type
            </th>
            <td>
              eth
            </td>
          </tr>
          <tr>
            <th>
              Bus
            </th>
            <td>
              eth
            </td>
          </tr>
          <tr>
            <th>
              Link speed
            </th>
            <td>
              N/A
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-lg-3"
          type="submit"
        >
           
           
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Interfaces/> Snapshot select interface. 1`] = `
<div>
  <h1>
    Network Interfaces
  </h1>
  <p>
    
Here you can configure the settings of the network interfaces on your device. You can switch the physical
network interfaces among networks. If you are unsure what to set here use the default settings.
        
  </p>
  <div
    class="col-sm-12 offset-lg-1 col-lg-10 p-0 mb-3"
  >
    <form>
      <h2>
        WAN
      </h2>
      <p>
        
It acts as an external network connection. Firewall rules should be applied here. It can only contain a
single interface.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                />
              </span>
            </div>
            <h5>
              WAN
            </h5>
          </button>
        </div>
      </div>
      <h2>
        LAN
      </h2>
      <p>
        
It acts as a local network connection. LAN should contain devices which are under your control and you
trust them. These devices can see each other and can access this web interface. It is recommended that the
LAN should contain at least one interface otherwise you might not be able to configure this device in an
easy way.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN3
            </h5>
          </button>
        </div>
      </div>
      <h2>
        Guest Network
      </h2>
      <p>
        
It acts as a local network connection. Unlike LAN the devices in the guest network can't access
the configuration interface of this device and are only able to access WAN (internet). This network should
be used for devices which you don't fully trust. Note that you can also limit download/upload speed of the
devices connected to the guest network.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface interface-selected"
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN1
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN4
            </h5>
          </button>
        </div>
      </div>
      <h2>
        Unassigned
      </h2>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN0
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN2
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div
              style="color: gray;"
            >
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="fas fa-wifi fa-stack-1x"
                />
                <i
                  class="fas fa-slash fa-stack-1x fa-inverse"
                />
                <i
                  class="fas fa-slash fa-stack-1x"
                  style="bottom: 0.3rem;"
                />
              </span>
            </div>
            <h5>
              1
            </h5>
          </button>
        </div>
      </div>
      <h3>
        Interface %s %s
      </h3>
      <div
        class="form-group"
      >
        <label
          for="1"
        >
          Network
        </label>
        <select
          class="custom-select"
          id="1"
        >
          <option
            value="lan"
          >
            LAN
          </option>
          <option
            value="guest"
          >
            Guest Network
          </option>
          <option
            value="none"
          >
            Unassigned
          </option>
        </select>
      </div>
      <table
        class="table table-hover"
      >
        <tbody>
          <tr>
            <th>
              State
            </th>
            <td>
              <span>
                <i
                  class="fa fa-lg fa-times-circle text-danger"
                  title="down"
                />
              </span>
            </td>
          </tr>
          <tr>
            <th>
              Module ID
            </th>
            <td>
              0
            </td>
          </tr>
          <tr>
            <th>
              Slot
            </th>
            <td>
              LAN1
            </td>
          </tr>
          <tr>
            <th>
              Interface ID
            </th>
            <td>
              lan1
            </td>
          </tr>
          <tr>
            <th>
              Type
            </th>
            <td>
              eth
            </td>
          </tr>
          <tr>
            <th>
              Bus
            </th>
            <td>
              eth
            </td>
          </tr>
          <tr>
            <th>
              Link speed
            </th>
            <td>
              N/A
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-lg-3"
          type="submit"
        >
           
           
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Interfaces/> Snapshot. 1`] = `
<div>
  <h1>
    Network Interfaces
  </h1>
  <p>
    
Here you can configure the settings of the network interfaces on your device. You can switch the physical
network interfaces among networks. If you are unsure what to set here use the default settings.
        
  </p>
  <div
    class="col-sm-12 offset-lg-1 col-lg-10 p-0 mb-3"
  >
    <form>
      <h2>
        WAN
      </h2>
      <p>
        
It acts as an external network connection. Firewall rules should be applied here. It can only contain a
single interface.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                />
              </span>
            </div>
            <h5>
              WAN
            </h5>
          </button>
        </div>
      </div>
      <h2>
        LAN
      </h2>
      <p>
        
It acts as a local network connection. LAN should contain devices which are under your control and you
trust them. These devices can see each other and can access this web interface. It is recommended that the
LAN should contain at least one interface otherwise you might not be able to configure this device in an
easy way.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN3
            </h5>
          </button>
        </div>
      </div>
      <h2>
        Guest Network
      </h2>
      <p>
        
It acts as a local network connection. Unlike LAN the devices in the guest network can't access
the configuration interface of this device and are only able to access WAN (internet). This network should
be used for devices which you don't fully trust. Note that you can also limit download/upload speed of the
devices connected to the guest network.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN1
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN4
            </h5>
          </button>
        </div>
      </div>
      <h2>
        Unassigned
      </h2>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN0
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN2
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div
              style="color: gray;"
            >
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="fas fa-wifi fa-stack-1x"
                />
                <i
                  class="fas fa-slash fa-stack-1x fa-inverse"
                />
                <i
                  class="fas fa-slash fa-stack-1x"
                  style="bottom: 0.3rem;"
                />
              </span>
            </div>
            <h5>
              1
            </h5>
          </button>
        </div>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-lg-3"
          type="submit"
        >
           
           
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Interfaces/> with various modules Snapshot select interface. 1`] = `
<div>
  <h1>
    Network Interfaces
  </h1>
  <p>
    
Here you can configure the settings of the network interfaces on your device. You can switch the physical
network interfaces among networks. If you are unsure what to set here use the default settings.
        
  </p>
  <div
    class="col-sm-12 offset-lg-1 col-lg-10 p-0 mb-3"
  >
    <form>
      <h2>
        WAN
      </h2>
      <p>
        
It acts as an external network connection. Firewall rules should be applied here. It can only contain a
single interface.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                />
              </span>
            </div>
            <h5>
              WAN
            </h5>
          </button>
        </div>
      </div>
      <h2>
        LAN
      </h2>
      <p>
        
It acts as a local network connection. LAN should contain devices which are under your control and you
trust them. These devices can see each other and can access this web interface. It is recommended that the
LAN should contain at least one interface otherwise you might not be able to configure this device in an
easy way.
        
      </p>
      <div
        class="network mb-3"
      >
        <div>
          <h4>
            Base module
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface "
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                2
              </h5>
            </button>
          </div>
        </div>
        <div>
          <h4>
            Module %s
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface "
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                2
              </h5>
            </button>
          </div>
        </div>
      </div>
      <h2>
        Guest Network
      </h2>
      <p>
        
It acts as a local network connection. Unlike LAN the devices in the guest network can't access
the configuration interface of this device and are only able to access WAN (internet). This network should
be used for devices which you don't fully trust. Note that you can also limit download/upload speed of the
devices connected to the guest network.
        
      </p>
      <div
        class="network mb-3"
      >
        <div>
          <h4>
            Base module
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface "
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                1
              </h5>
            </button>
          </div>
        </div>
        <div>
          <h4>
            Module %s
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface interface-selected"
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                1
              </h5>
            </button>
          </div>
        </div>
      </div>
      <h2>
        Unassigned
      </h2>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN0
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div
              style="color: gray;"
            >
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="fas fa-wifi fa-stack-1x"
                />
                <i
                  class="fas fa-slash fa-stack-1x fa-inverse"
                />
                <i
                  class="fas fa-slash fa-stack-1x"
                  style="bottom: 0.3rem;"
                />
              </span>
            </div>
            <h5>
              1
            </h5>
          </button>
        </div>
      </div>
      <h3>
        Interface %s %s
      </h3>
      <div
        class="form-group"
      >
        <label
          for="1"
        >
          Network
        </label>
        <select
          class="custom-select"
          id="1"
        >
          <option
            value="lan"
          >
            LAN
          </option>
          <option
            value="guest"
          >
            Guest Network
          </option>
          <option
            value="none"
          >
            Unassigned
          </option>
        </select>
      </div>
      <table
        class="table table-hover"
      >
        <tbody>
          <tr>
            <th>
              State
            </th>
            <td>
              <span>
                <i
                  class="fa fa-lg fa-times-circle text-danger"
                  title="down"
                />
              </span>
            </td>
          </tr>
          <tr>
            <th>
              Module ID
            </th>
            <td>
              1
            </td>
          </tr>
          <tr>
            <th>
              Slot
            </th>
            <td>
              1
            </td>
          </tr>
          <tr>
            <th>
              Interface ID
            </th>
            <td>
              lan4
            </td>
          </tr>
          <tr>
            <th>
              Type
            </th>
            <td>
              eth
            </td>
          </tr>
          <tr>
            <th>
              Bus
            </th>
            <td>
              eth
            </td>
          </tr>
          <tr>
            <th>
              Link speed
            </th>
            <td>
              N/A
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-lg-3"
          type="submit"
        >
           
           
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Interfaces/> with various modules Snapshot. 1`] = `
<div>
  <h1>
    Network Interfaces
  </h1>
  <p>
    
Here you can configure the settings of the network interfaces on your device. You can switch the physical
network interfaces among networks. If you are unsure what to set here use the default settings.
        
  </p>
  <div
    class="col-sm-12 offset-lg-1 col-lg-10 p-0 mb-3"
  >
    <form>
      <h2>
        WAN
      </h2>
      <p>
        
It acts as an external network connection. Firewall rules should be applied here. It can only contain a
single interface.
        
      </p>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                />
              </span>
            </div>
            <h5>
              WAN
            </h5>
          </button>
        </div>
      </div>
      <h2>
        LAN
      </h2>
      <p>
        
It acts as a local network connection. LAN should contain devices which are under your control and you
trust them. These devices can see each other and can access this web interface. It is recommended that the
LAN should contain at least one interface otherwise you might not be able to configure this device in an
easy way.
        
      </p>
      <div
        class="network mb-3"
      >
        <div>
          <h4>
            Base module
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface "
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                2
              </h5>
            </button>
          </div>
        </div>
        <div>
          <h4>
            Module %s
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface "
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                2
              </h5>
            </button>
          </div>
        </div>
      </div>
      <h2>
        Guest Network
      </h2>
      <p>
        
It acts as a local network connection. Unlike LAN the devices in the guest network can't access
the configuration interface of this device and are only able to access WAN (internet). This network should
be used for devices which you don't fully trust. Note that you can also limit download/upload speed of the
devices connected to the guest network.
        
      </p>
      <div
        class="network mb-3"
      >
        <div>
          <h4>
            Base module
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface "
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                1
              </h5>
            </button>
          </div>
        </div>
        <div>
          <h4>
            Module %s
          </h4>
          <div
            class="scrollable"
          >
            <button
              class="interface "
              type="button"
            >
              <div>
                <span
                  class="fa-stack fa-2x"
                >
                  <i
                    class="far fa-square fa-stack-2x"
                  />
                  <i
                    class="fas fa-ethernet fa-stack-1x"
                    style="color: lightgrey;"
                  />
                </span>
              </div>
              <h5>
                1
              </h5>
            </button>
          </div>
        </div>
      </div>
      <h2>
        Unassigned
      </h2>
      <div
        class="network mb-3"
      >
        <div
          class="scrollable"
        >
          <button
            class="interface "
            type="button"
          >
            <div>
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="far fa-square fa-stack-2x"
                />
                <i
                  class="fas fa-ethernet fa-stack-1x"
                  style="color: lightgrey;"
                />
              </span>
            </div>
            <h5>
              LAN0
            </h5>
          </button>
          <button
            class="interface "
            type="button"
          >
            <div
              style="color: gray;"
            >
              <span
                class="fa-stack fa-2x"
              >
                <i
                  class="fas fa-wifi fa-stack-1x"
                />
                <i
                  class="fas fa-slash fa-stack-1x fa-inverse"
                />
                <i
                  class="fas fa-slash fa-stack-1x"
                  style="bottom: 0.3rem;"
                />
              </span>
            </div>
            <h5>
              1
            </h5>
          </button>
        </div>
      </div>
      <div
        class="text-right"
      >
        <button
          class="btn btn-primary  col-sm-12 col-lg-3"
          type="submit"
        >
           
           
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
